lib/features/composer/presentation/composer_controller.dart

```diff
@@ -1412,9 +1418,30 @@ class ComposerController extends BaseController with DragDropFileMixin implement
 
     if (composerArguments.value?.emailActionType == EmailActionType.editDraft) {
       _setUpRequestReadReceiptForDraftEmail(success.emailCurrent);
+      _restoreIdentityFromHeader(success.emailCurrent);
+      _restoreCollapsibleButton(success.emailCurrent);
     }
   }
 
+  void _restoreIdentityFromHeader(Email? email) {
+    final identityIdFromHeader = email?.identityIdFromHeader;
+    if (identityIdFromHeader == null) return;
+    final selectedIdentityFromHeader = _selectedIdentityFromId(identityIdFromHeader);
+    if (selectedIdentityFromHeader == null) return;
+    identitySelected.value = selectedIdentityFromHeader;
+  }
+
+  Future<void> _restoreCollapsibleButton(Email? email) async {
+    final emailContent = email?.emailContentList.asHtmlString;
+    if (emailContent == null) return;
+    final emailDocument = parse(emailContent);
+    final signature = emailDocument.querySelector('div.tmail-signature');
+    if (signature == null) return;
+    await Future.delayed(const Duration(milliseconds: 300), () async {
+      await _applySignature(signature.innerHtml);
+    });
```

Comment:
We should use `try/catch` block at here.

Reply:
Done